### PR TITLE
SB-536: Implement cron tasks

### PR DESCRIPTION
**Pull request by ypkkhatri**
- Url context restored to /configuration/
- Swagger annotations applied
- Changes made in CronTaskClient

### CODE REVIEW

File: strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/client/CronTaskClient.java [hunk 1]
```diff
@@ -21,11 +21,11 @@
  * @author mtodorov
  * @author Yougeshwar
  */
-public class TestClient
+public class CronTaskClient
```
Comment (carlspring) [inline strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/client/CronTaskClient.java]: Could you please make this extend [`ArtifactClient`](https://github.com/strongbox/strongbox/blob/master/strongbox-client/src/main/java/org/carlspring/strongbox/client/ArtifactClient.java), like we'd discussed? Could you also remove all the methods which are already present in it?

File: strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/crontask/rest/CronTaskConfigurationRestletTest.java [hunk 2]
```diff
@@ -1,9 +1,10 @@
 package org.carlspring.strongbox.crontask.rest;
 
+import org.carlspring.strongbox.client.CronTaskClient;
 import org.carlspring.strongbox.crontask.domain.CronTaskConfiguration;
 import org.carlspring.strongbox.config.CronTasksConfig;
 import org.carlspring.strongbox.crontask.exceptions.CronTaskNotFoundException;
-import org.carlspring.strongbox.crontask.test.MyTask;
+import org.carlspring.strongbox.crontask.tasks.MyTask;
```
Comment (carlspring) [inline strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/crontask/rest/CronTaskConfigurationRestletTest.java]: @ypkkhatri : Could you please rename the `org.carlspring.strongbox.crontask` to just `org.carlspring.strongbox.cron`. Also, move the `MyTask` class under it.

File: strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/client/CronTaskClient.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        setupAuthentication(resource);
+
+        Response response = resource.request().delete();
+
+        handleFailures(response, "Failed to delete artifact!");
+
+        return response;
+    }
+
+    private void handleFailures(Response response, String message)
```
Comment (carlspring) [inline strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/client/CronTaskClient.java]: @ypkkhatri : This method already exists in the parent class, if I remember correctly.

Comment (carlspring) [inline strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/client/CronTaskClient.java]: @ypkkhatri : It also doesn't look like you've extracted the various methods from the test classes that issue requests, as separate methods in the client, like we'd discussed. Would you mind looking into this, please?

Comment (ypkkhatri) [inline strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/client/CronTaskClient.java]: That method is private in parent class so unable too access it, that's why created in this class

Comment (ypkkhatri) [inline strongbox-cron-tasks-rest/src/test/java/org/carlspring/strongbox/client/CronTaskClient.java]: extract various methods I didn't get that, those methods already exists in parent class then I should override them? I didn't find generic delete method for all in it, that why create in this class, and can you please elaborate what you need more?

**Pull request merged**